### PR TITLE
[FEATURE] Improve Reddit sources to use global search endpoint #1513

### PR DESCRIPTION
**Pull request by lcaresia**
Added a action `search-post`, need test too.

### CODE REVIEW

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible

File: components/reddit/reddit.app.mjs [hunk 1]
```diff
@@ -216,11 +231,12 @@ export default {
...
-      408,
-      429,
-      500,
-    ].includes(statusCode);
+    _isRetriableStatusCode(statusCode) {
+      [
+        408,
+        429,
+        500,
+      ].includes(statusCode);
```
Comment (jcortes) [inline components/reddit/reddit.app.mjs:239]:
```suggestion
      return [
        408,
        429,
        500,
      ].includes(statusCode);
```

**Review by jcortes — CHANGES_REQUESTED** [review]
HI @lcaresia I've added just one minor fix and it'll be ready for QA

**Review by jcortes — COMMENTED** [review]
Hi @lcaresia LGTM. Ready for QA!

**Comment by vunguyenhung** [comment]
Hi everyone, all test cases are passed! Ready for release! 

Test report
[Reddit_1513.pdf](https://github.com/PipedreamHQ/pipedream/files/8949538/Reddit_1513.pdf)

**Review by michelle0927 — APPROVED** [review]

**Pull request merged**